product-service/productById.js

```diff
@@ -1,24 +1,34 @@
 'use strict';
 
-const AWS = require('aws-sdk');
-const s3 = new AWS.S3();
+const { Client } = require('pg');
+const { PG_HOST, PG_PORT, PG_DATABASE, PG_USERNAME, PG_PASSWORD } = process.env;
+const dbParams = {
+  host: PG_HOST,
+  port: PG_PORT,
+  database: PG_DATABASE,
+  user: PG_USERNAME,
+  password: PG_PASSWORD,
+  ssl: {
+    rejectUnauthorized: false
+  },
+  connectionTimeoutMillis: 5000
+}
 
-const bucketName = process.env.S3BucketName;
-const productListFileName = process.env.productListFileName;
+module.exports.getProductById = async event => {
 
+  console.log(`Request: ${JSON.stringify(event)}`);
+  const pathParam = event.pathParameters.id;
+  const query = `select id, title, description, price, count 
+  from products p join stocks s on p.id = s.product_id
+  where p.id = '${pathParam}'`;
 
-module.exports.getProductById = async event => {
-  const params = {
-    Bucket: bucketName,
-    Key: productListFileName
-  }
+  const client = new Client(dbParams);
+  await client.connect();
 
   try {
-    const file = await s3.getObject(params).promise();
-    const productList = JSON.parse(file.Body.toString('utf8'));
-    const pathParam = event.pathParameters.id;
-    const product = productList.filter(product => product.id === pathParam)[0];
-    if (product === undefined) {
+    const { rows: product } = await client.query(query);
+    console.log(`Result: ${JSON.stringify(product)}`);
+    if (product[0] === undefined) {
       return {
```

Comment:
This exeption is not working (https://lbxm5xv6d2.execute-api.eu-central-1.amazonaws.com/dev/products/dddd check here)

Reply:
Thank you for response. You should use uuid format for requesting product. In case wrong uuid 404 error will be thrown